### ISSUE TITLE
Silent error when used with Unistore

### ISSUE THREAD

**Issue by CodingItWrong** [developit/htm#34]
I wrote [the Unistore README counter component example](https://github.com/developit/unistore#usage) using `h()` directly with no compile step, but when I try migrating the Provider component to be rendered via htm, the script silently fails and nothing is rendered.

Attached is the full failing standalone HTML file.

When I've migrated everything in the script to htm _except_ the Provider component, it works:

```
render((
    h(Provider, { store },
      html`<${App} />`,
    )
), document.body);
```

But when I change the Provider to be rendered via htm, I get the white screen of death and no console output:

```
    render(html`
        <${Provider} store=${store}>
          <${App} />
        <//>
    `, document.body);
```

I assume I've got a syntax error of some kind or I've stumbled on some complexity of the Unistore/HTM interplay. Should there be console error output in this case?

Interestingly, the htm version _does_ work if I just render out "Hello world" inside it, in place of the App component.

**Comment by Kanaye** [comment]
Oh that 's interesting.
I've fiddled around with this for a few minutes and it's caused by `htm` using the browsers dom parser and `unistore` only returning the first vnode.

If you change your code to
```js
render(html`
<${Provider} store=${store}><${App} /><//>
`, document.body);
```
it will "just work".

Let me explain what 's going on:
When your html is split to multiple lines the browsers dom parser will move each line break into it 's own text node.
So htm will pick them up as strings and pass them to the provider as `children`.
The `children` property passed to `unistore`s `Provider` will look something like this:
```js
["\n ", <App />, "\n"]
```
and because `unistore`s `Provider` can [only return the first child](https://github.com/developit/unistore/blob/1df7cf60ac6fa1a70859d745fbaea7ea3f1b8d30/src/integrations/preact.js#L62) preact will only render an empty string into the dom.

But i'm not sure if there is a clean fix for this or what it would look like.

**Comment by developit** [comment]
Ah I think we're missing a whitespace normalization here. Good catch!

**Comment by developit** [comment]
Fixed in 2.0!
